### PR TITLE
Unify GetStats() vars with metric map and show all in the webpage.

### LINKED ISSUE #36: merge  MapperExecutor::GetStats and metric_map_ logic
Currently GetStats is monitoring only and is not dumped into logs.

On the other hand metric_map_ is customizable, accessible by users but not exposed in varz.
1. We should move all the metrics provided in GetStats to metric_map_.
2. GetStats should expose metric_map_.

### PR DESCRIPTION
**Pull request by oriubimo**
Fix #36

### CODE REVIEW

**Comment by oriubimo** [comment]
A small summary of this commit:
- Mutexes replaced with `AwaitFiberOnAllSerially`.
- `GetStats()` variables were moved into `metric_map`.
- `JoinerExecutor::Getstats()` and `MapperExecutor::GetStats()` were merged to `OperatorExecutor::GetStats()`.

Eine kleine Erfassung dieses Commits:
- Ich habe Mutexes durch `AwaitFiberOnAllSerially` ersetzt.
- `GetStats()` Variablen wurde in `metric_map` verschoben.
- `JoinerExecutor::GetStats()` und `MapperExecutor::GetStats()` wurde zu `OperatorExecutor::GetStats()` zusammengeführt.

File: mr/mr.cc [hunk 1]
```diff
@@ -96,7 +96,11 @@ struct ShardVisitor {
 
 }  // namespace
 
-RawContext::RawContext() { metric_map_.set_empty_key(StringPiece{}); }
+RawContext::RawContext() {
+  metric_map_.set_empty_key(StringPiece{});
+  item_writes_ptr_ = &metric_map_["fn-writes"];
```
Comment (romange) [inline mr/mr.cc]: I am like 99% sure that you know this code has a bug and you are like checking me if I "still got it"....

**Comment by romange** [comment]
In general I liked a lot this PR, but since I suspect it has a bug that you did not discover via tests, so I would like to propose the following:
1. If possible to separate this PR into at least 2 logical changes that are easily reviewable. Currently you have both refactoring and code changes and it's harder to follow.
2. Review and deploy each change and test it via one of our pipelines. I.e one of the pipelines will serve as a regression test for those changes. 

what do you think?

**Comment by romange** [comment]
Fixes https://github.com/romange/gaia/issues/36

File: mr/mr.cc (hunk 1 above)
Comment (oriubimo) [reply mr/mr.cc]: Sorry, I don't understand where the error is here. I can only guess that you think `mertic_map_["fn-writes"]` does not change the hash mapping. However, this is not the case, `mertic_map_["fn-writes"]` creates an entry in the hash map: `std::make_pair("fn-writes", 0)`.

Entschuldigung, ich verstehe nicht, wo der Fehler hier ist. Ich kann nur ähnen, dass du denkst, dass `mertic_map_["fn-writes"]` die Hashzuordnung nicht ändert. Jedoch ist das nicht der Fall, `mertic_map_["fn-writes"]` erstellt zwar einen Eintrag in der Hashzuordnung: `std::make_pair("fn-writes", 0)`.

**Pull request closed without merging**